### DISCUSSION TITLE
Get the commit ID / revision before doing a push/commint

### DISCUSSION THREAD

**Discussion by alexvagtm** [community/community#103128] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello, 

Previous to GIT i was using a way of inserting the revision id in a document.

File1 Current Version 1.2 

I made some changes and mentioned in it that now it is version 1.3 
then do the check-in.

With Git I can't do that/or I don't know how to do that as the commit id is generated only after the commit.

Any idea how to get it prior to the commit, like to get the next commit id that I will create.

Thank you

**Comment by gil-son** [comment]
In Git, you can't know the commit hash before actually making the commit. The commit hash is generated based on the content of the commit, including the changes made and the current state of the repository. It is a SHA-1 hash of the commit content.

If you need a version identifier prior to making a commit, you might consider using a branch or a tag to mark the state of your repository. For example:

Create a branch for the new version:
```
git branch version-1.3
git checkout version-1.3
```

Make your changes and commit them:

```
git add .
git commit -m "Version 1.3"
```

At this point, you won't know the actual commit hash, but you have a branch pointing to the commit.
Continue working on the new version:

```
 git add .
 git commit -m "Some additional changes for Version 1.3"
```

When you are ready to finalize the version, you can merge the branch back into your main branch (e.g., main or master). The commit hash will be generated at that point.

```
git checkout main
git merge version-1.3
```
